zebra-grpc/src/server.rs

```diff
@@ -170,34 +270,74 @@ where
             ));
         };
 
-        // If there are no results for a key, we still want to return it with empty results.
-        let empty_map = BTreeMap::new();
-
-        let results = keys
-            .into_iter()
-            .map(|key| {
-                let values = response.get(&key).unwrap_or(&empty_map);
-
-                // Skip heights with no transactions, they are scanner markers and should not be returned.
-                let transactions = Results {
-                    transactions: values
-                        .iter()
-                        .filter(|(_, transactions)| !transactions.is_empty())
-                        .map(|(height, transactions)| {
-                            let txs = transactions.iter().map(ToString::to_string).collect();
-                            (height.0, TransactionHash { hash: txs })
-                        })
-                        .collect(),
-                };
-
-                (key, transactions)
-            })
-            .collect::<BTreeMap<_, _>>();
+        let results = process_results(keys, response);
 
         Ok(Response::new(GetResultsResponse { results }))
     }
 }
 
+fn process_results(
```

Comment:
Optional:
Can we use the function in https://github.com/ZcashFoundation/zebra/blob/main/zebra-grpc/src/server.rs#L176 ?

Reply:
Do you mean to use this function instead of the closure?